cli/src/commands/rebase.rs

```diff
@@ -290,13 +290,27 @@ fn rebase_descendants(
     rebase_options: RebaseOptions,
 ) -> Result<(), CommandError> {
     workspace_command.check_rewritable(old_commits)?;
+    let (skipped_commits, old_commits) = old_commits
+        .iter()
+        .partition::<Vec<_>, _>(|commit| commit.parents().eq(new_parents));
+    for commit in skipped_commits.iter() {
+        writeln!(
+            ui.stderr(),
+            "Skipping rebase of commit {}",
+            short_commit_hash(commit.id()),
+        )?;
+        continue;
+    }
+    if old_commits.is_empty() {
+        return Ok(());
```

Comment:
nit: "Nothing changed." could be printed here

Reply:
I wasn't sure about what output we wanted to show actually. I shifted some stuff outside of transactions (and duplicated a check) since I didn't want to create a new transaction, which would also show the "Nothing changed." message when the transaction finishes. Should I just add the message manually, or just create a transaction even if the rebase is a no-op and let the transaction finish method print the message?

Reply:
I thought printing "Nothing changed" is good because it's unusual situation, but we can also say it's redundant because skipped revisions are printed one by one.

> Should I just add the message manually, or just create a transaction even if the rebase is a no-op and let the transaction finish method print the message?

I would use simpler method. FWIW, in some places, we check `tx.mut_repo().has_changes()` manually to silence "Nothing changed.". That's another way around (if printing "Nothing changed" seemed too verbose.)

Reply:
I went with silencing "Nothing changed" for now, since I feel that the messages for rebase of each commit being skipped are sufficient. Happy to revert if anyone feels more strongly about it.